app/src/main/java/com/example/habitsmasher/HabitEventList.java

```diff
@@ -179,4 +192,55 @@ private void setHabitEventDataInDatabase(String userId, Habit parentHabit, Strin
                 .addOnSuccessListener(new SuccessListener(TAG, "Data successfully added."))
                 .addOnFailureListener(new FailureListener(TAG, "Data failed to be added."));
     }
+
+    /**
+     * @param image The image to add to the database
+     * @param id The ID of the habit event corresponding to the image
+     */
+    public void addImageToDatabase(String userId, Habit parentHabit, Uri image, String id) {
+        Uri toAdd = image;
+
+        if (toAdd == null) {
+            image = Uri.parse("android.resource://com.example.habitsmasher/drawable/habit_temp_img");
+
+        }
+        // Get firebase storage
+        FirebaseStorage storage = FirebaseStorage.getInstance();
+        StorageReference storageReference = storage.getReference();
```

Comment:
I see these are fetched multiple times in different methods, can we make these fields instead?